src/plugins/intel_cpu/src/nodes/executors/acl/acl_executor.hpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "cpu_memory.h"
+#include "nodes/executors/executor.hpp"
+#include "arm_compute/runtime/NEON/NEFunctions.h"
+
+namespace ov {
+namespace intel_cpu {
+
+using ACLMemoryArgs     = std::unordered_map<int, std::shared_ptr<arm_compute::Tensor>>;
+using ACLMemoryInfoArgs = std::unordered_map<int, std::shared_ptr<arm_compute::TensorInfo>>;
+using ACLFunction       = std::unique_ptr<arm_compute::IFunction>;
+
+struct ACLTensorAttrs {
+    bool enableNHWCReshape = false;
+    size_t maxDimsShape = arm_compute::MAX_DIMS;
+};
+
+class ACLCommonExecutor : public Executor {
+public:
+    virtual void prepareTensorsInfo() {
+        OPENVINO_THROW_NOT_IMPLEMENTED("This version of the 'prepareTensorsInfo' method is not implemented by executor");
+    }
+    virtual void configureFunction() {
+        OPENVINO_THROW_NOT_IMPLEMENTED("This version of the 'configureFunction' method is not implemented by executor");
+    }
+    impl_desc_type implType() const override {
+        return impl_desc_type::acl;
+    }
+    void execute(const MemoryArgs& memory) override;
+    bool update(const MemoryArgs& memory) override;
+
+protected:
+    arm_compute::Status tensorsInfoValidateStatus;
+    ACLFunction iFunction = nullptr;
+    ACLMemoryArgs aclMemoryArgs;
+    ACLMemoryInfoArgs aclMemoryInfoArgs;
```

Comment:
Is `TensorInfo` available from `Tensor`: `ITensorInfo Tensor::info()`? If yes, then we don't need this class member.

Reply:
@eshoguli `Tensor` doesn't exists in `prepareTensorsInfo` function and only after this function we can create `Tensor`

Reply:
ok, thanks